tests/scripts/perf-sanity/run_benchmark_serve.py

```
@@ -0,0 +1,370 @@
+            
+            # Wait for server to be ready
+            if not self.wait_for_server(server_process.pid):
+                print("Failed to start server, killing process and skipping this test case")
+                try:
+                    subprocess.run(f"kill -9 {server_process.pid}", shell=True, check=False)
+                    subprocess.run(f"wait {server_process.pid} 2>/dev/null || true", shell=True, check=False)
+                except Exception as e:
+                    print(f"Warning: Error during server cleanup: {e}")
+                return
+            
+            # Run all benchmarks for this test case
```

Ruff (0.12.2)

Line too long (124 > 120)

(E501)